toolchain/check/testdata/facet/access.carbon

```diff
@@ -80,6 +80,55 @@ fn UseIndirect[T:! I](x: T) -> T {
 //@dump-sem-ir-end
 }
 
+// --- fail_todo_access_through_call.carbon
+library "[[@TEST_NAME]]";
+
+interface I {
+  let X:! type;
+  fn G() -> X;
+}
+
+fn F(U:! I where .X = .Self) {
+  U.G().G();
+
+  // TODO: This should typecheck.
+  // - The first `.` is on a NameRef of type FacetType for `I where .X = .Self`.
+  //   - This finds `G` through the FacetType.
+  // - The second `.` is on a Call of type FacetAccessType into `BindSymbolicName` with type FacetType for `I`.
+  //   - This finds `G` through the FacetType (impl lookup strips off FacetAccessType).
+  // - The second `.` is on a Call of type FacetAccessType into `ImplWitnessAccess` of `I.X` into `LookupImplWitness`, which has type `type`
+  //   - Can't make calls on an `ImplWitnessAccess`.
```

Comment:
FWIW, I would expect that the constant value of the `ImplWitnessAccess` would be the same type that we got for the second lookup.